text/2018-10-25-tikv-client-rust.md

```diff
@@ -0,0 +1,386 @@
+
+    fn delete_range<R, C>(&self, range: R, cf: C) -> KvFuture<()>
+    where
+        R: Into<KeyRange>,
+        C: Into<Option<String>>;
+```
+
+#### Usage example of the Raw Key-Value API
+
+```rust
+extern crate futures;
+extern crate tikv_client;
+
+use futures::future::Future;
+use tikv_client::raw::Client;
+use tikv_client::*;
+
+fn main() {
+    let config = Config::new(vec!["127.0.0.1:3379"]);
+    let raw = raw::RawClient::new(&config)
+        .wait()
+        .expect("Could not connect to tikv");
+
+    let key: Key = b"Company".to_vec().into();
+    let value: Value = b"PingCAP".to_vec().into();
+
+    raw.put((Clone::clone(&key), Clone::clone(&value)), None)
+        .wait()
+        .expect("Could not put kv pair to tikv");
+    println!("Successfully put {:?}:{:?} to tikv", key, value);
+
+    let value = raw
+        .get(Clone::clone(&key), None)
+        .wait()
+        .expect("Could not get value");
+    println!("Found val: {:?} for key: {:?}", value, key);
+
+    raw.delete(Clone::clone(&key), None)
+        .wait()
+        .expect("Could not delete value");
+    println!("Key: {:?} deleted", key);
+
+    raw.get(key, None)
+        .wait()
+        .expect_err("Get returned value for not existing key");
+}
+```
+
+The result is like:
+
+```bash
+Successfully put Key([67, 111, 109, 112, 97, 110, 121]):Value([80, 105, 110, 103, 67, 65, 80]) to tikv
+Found val: Value([80, 105, 110, 103, 67, 65, 80]) for key: Key([67, 111, 109, 112, 97, 110, 121])
+Key: Key([67, 111, 109, 112, 97, 110, 121]) deleted
+```
+
+Raw Key-Value client is a client of the TiKV server and only supports the GET/BATCH_GET/PUT/BATCH_PUT/DELETE/BATCH_DELETE/SCAN/BATCH_SCAN/DELETE_RANGE commands. The Raw Key-Value client can be safely and concurrently accessed by multiple threads. Therefore, for one process, one client is enough generally.
+
+### Try the Transactional Key-Value API
+
+The Transactional Key-Value API is more complicated than the Raw Key-Value API. Some transaction related concepts are listed as follows.
+
+- Client
+
+    Like the Raw Key-Value client, a Client is client to a TiKV cluster.
+
+- Snapshot
+
+    A Snapshot is the state of a Client at a particular point of time, which provides some readonly methods. The multiple reads of the same Snapshot is guaranteed consistent.
+
+- Transaction
+
+    Like the Transaction in SQL, a Transaction symbolizes a series of read and write operations performed within the Client. Internally, a Transaction consists of a Snapshot for reads, and a buffer for all writes. The default isolation level of a Transaction is Snapshot Isolation.
+
+To use the Transactional Key-Value API, take the following steps:
+
+1. Create an instance of Config to specify endpoints of PD (Placement Driver) and optional security config
+
+```rust
+    let config = Config::new(vec!["127.0.0.1:2379"]);
+```
+
+2. Create a Transactional Key-Value client.
+
+```rust
+    let client = TxnClient::new(&config);
+```
```

Comment:
Please also indent the code blocks in step 1&2 like that in step 5.

Reply:
Fixed, but I'm not sure if I'm doing it right, please double check it.